### PR TITLE
Multi-logger handling

### PR DESCRIPTION
**Pull request by awelzel**
If there are multiple loggers configured in node.cfg, currently they all invoke
archive-log for their own logs and overwrite each others files during log rotation
due to having the same rotation intervals and creating the same names via
`make-archive-name`. There's no easy way to customize the name for individual
loggers.

This PR proposes the following API/interface:

* Invoke the rotation postprocessor with a new environment variable
  called ZEEK_ARG_LOG_SUFFIX. This environment variable is set *only*
  when multiple loggers are configured. It's set to the value of
  Cluster::node.

* Place a .log_suffix file within a logger's working directory also
  *only* when multiple loggers are configured. This can be used by
  the post-terminate script to set the ZEEK_ARG_LOG_SUFFIX to set it
  for the archive-log / make-archive-name.

* Make the `make-archive-name` and `post-terminate` ZEEK_ARG_LOG_SUFFIX
  and .log_suffix aware.

The result is that the name of archived logs includes the logger name
suffix when multiple loggers are configured. This is configurable using
a custom `make-archive-name` script.

    -rw-rw-r-- 1 zeek zeek 8.7K Apr  6 11:58 conn.11:57:00-11:58:00-logger-1.log.gz
    -rw-rw-r-- 1 zeek zeek 8.7K Apr  6 11:58 conn.11:57:00-11:58:00-logger-2.log.gz
    -rw-rw-r-- 1 zeek zeek  529 Apr  6 11:58 conn-summary.11:57:00-11:58:00-logger-1.log.gz
    -rw-rw-r-- 1 zeek zeek  367 Apr  6 11:58 conn-summary.11:57:00-11:58:00-logger-2.log.gz

### CODE REVIEW

**Comment by awelzel** [comment]
@rsmmr - requesting a review from you as mentioned in DM. I've tried to keep it small.

The description of the PR has some background what this is attempting to do. What I mainly dislike a somewhat is:
* ZEEK_ARG_LOGGER_NODE only being set in the multi-logger scenario. It seems that may as well be valid in the single logger scenario and we could have a zeekctl-config.sh setting with the number of loggers, but that would make usage a bit more complicated ( same for the .logger_node magic file).
* Undecided about the new logging.zeek file, but there's some reliance on Zeek's base that I believe that would be nice to avoid (and vice-versa, Zeek "knowing" about the archive-log script.


I'm going to post on community.zeek.org and see if there's any input how people have so far solved this.

**Comment by rsmmr** [comment]
Look good to me in general. Couple quick thoughts:

> ZEEK_ARG_LOGGER_NODE only being set in the multi-logger scenario.

We could just make that `ZEEK_ARG_LOG_SUFFIX`, then it's clear it's the responsibility if the caller to decide whether to append something or not.

> Undecided about the new logging.zeek file, b

I think I'd update that in `base`. Might not be great there but it's what we currently have, so that would be the minimal change (and we're phasing out ZeekControl anyways)

Also, a test for the new functionality would be good (although I do realize you'd be the 1st person touching the zeekctl test suite in many years ...)

File: scripts/zeekctl/logging.zeek [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+global log_suffix_filename = ".log_suffix";
+
+@if ( Cluster::local_node_type() == Cluster::LOGGER )
+event zeek_init()
+	{
+	local logger_count = 0;
+	for ( _, n in Cluster::nodes )
+		if ( n$node_type == Cluster::LOGGER )
+			++logger_count;
```
Comment (rsmmr) [inline scripts/zeekctl/logging.zeek:18]: My personal readability preference would be:

```suggestion
	for ( _, n in Cluster::nodes )
	    {
		if ( n$node_type == Cluster::LOGGER )
			++logger_count;
		}
```

File: scripts/zeekctl/logging.zeek [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	if ( Cluster::local_node_type() == Cluster::LOGGER )
```
Comment (rsmmr) [inline scripts/zeekctl/logging.zeek:20]: Isn't this already covered by the `@if` (or vice versa)?

File: scripts/zeekctl/logging.zeek [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+			if ( ! active_file(f) )
+				{
+				Reporter::error(fmt("failed to open %s", log_suffix_filename));
+				return;
+				}
+
+			if ( ! write_file(f, fmt("%s\n", Cluster::node)) )
+				{
+				Reporter::error(fmt("failed to write %s", log_suffix_filename));
+				}
```
Comment (rsmmr) [inline scripts/zeekctl/logging.zeek:39]: Another personal preference:
```suggestion
				Reporter::error(fmt("failed to write %s", log_suffix_filename));
```

**Review by rsmmr — APPROVED** [review]
Looks good!

File: scripts/zeekctl/logging.zeek (hunk 2 above)
Comment (awelzel) [reply scripts/zeekctl/logging.zeek:20]: Yeah, I added the `@if` and forget to remove the other one.

I'm going to add test-cases to exercise this a bit, and also for `post-terminate` before merging, but thanks for looking through!

**Pull request merged**